Buildings/Fluid/HeatExchangers/ThermalWheels/Latent/BaseClasses/Effectiveness.mo

```diff
@@ -0,0 +1,163 @@
+    final unit="K",
+    displayUnit="degC")
+    "Exhaust air temperature"
+    annotation (Placement(transformation(extent={{-140,-100},{-100,-60}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput VSup_flow(final unit="m3/s")
+    "Supply air volumetric flow rate"
+    annotation (Placement(transformation(extent={{-140,60},{-100,100}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput VExh_flow(final unit="m3/s")
+    "Exhaust air volumetric flow rate"
+    annotation (Placement(transformation(extent={{-140,20},{-100,60}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput uSpe(
+    final unit="1",
+    final max=1)
+    "Wheel speed ratio"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput epsSen(final unit="1")
+    "Sensible heat exchanger effectiveness"
+    annotation (Placement(transformation(extent={{100,20},{140,60}}),
+        iconTransformation(extent={{100,10},{140,50}})));
+  Buildings.Controls.OBC.CDL.Interfaces.RealOutput epsLat(final unit="1")
+    "Latent heat exchanger effectiveness"
+    annotation (Placement(transformation(extent={{100,-60},{140,-20}}),
+        iconTransformation(extent={{100,-50},{140,-10}})));
+
+protected
+  Real rat
+    "Ratio of the average operating air flow rate to the nominal supply air flow rate";
+  Real epsSenPL
+    "Part load sensible heat exchanger effectiveness used for calculation";
+  Real epsSen_nominal
+    "Nominal sensible heat exchanger effectiveness used for calculation";
+  Real epsLatPL
+    "Part load latent heat exchanger effectiveness used for calculation";
+  Real epsLat_nominal
+    "Nominal latent heat exchanger effectiveness used for calculation";
+
+equation
+  // Check if the air flows are too unbalanced
+  assert(VSup_flow - 2*VExh_flow < 0 or VExh_flow - 2*VSup_flow < 0,
+    "In " + getInstanceName() + ": The ratio of the supply flow rate to the exhaust flow rate should be in the range of [0.5, 2].",
+    level=AssertionLevel.warning);
+  // Calculate the average volumetric air flow and flow rate ratio.
+  rat = (VSup_flow + VExh_flow)/2/VSup_flow_nominal;
+  // Check if the extrapolation goes too far
+  assert(rat > 0.5 and rat < 1.3,
+    "In " + getInstanceName() + ": The ratio of the operating flow rate to the nominal supply flow rate should be in the range of [0.5, 1.3].",
+    level=AssertionLevel.warning);
+  // Switch between cooling and heating modes based on the difference between the supply air temperature and the exhaust air temperature
+  epsSenPL = Buildings.Utilities.Math.Functions.regStep(TSup-TExh, epsSenCooPL, epsSenHeaPL, 1e-5);
+  epsSen_nominal = Buildings.Utilities.Math.Functions.regStep(TSup-TExh, epsSenCoo_nominal, epsSenHea_nominal, 1e-5);
+  epsLatPL = Buildings.Utilities.Math.Functions.regStep(TSup-TExh, epsLatCooPL, epsLatHeaPL, 1e-5);
+  epsLat_nominal = Buildings.Utilities.Math.Functions.regStep(TSup-TExh, epsLatCoo_nominal, epsLatHea_nominal, 1e-5);
```

Comment:
These 4 calls to `regStep` should be changed: There should only be one call to `regStep` as the first argument is always the same. 
In many configurations, the parameters for the efficiency are the same, in which case no `regStep` (or weighting of the two values) is needed.

Reply:
I changed these calls.